spec/tapioca/dsl/compilers/protobuf_spec.rb

```diff
@@ -664,6 +536,20 @@ def progress=(value); end
             end
           end
         end
+
+        private
+
+        sig { params(name: String, content: String, require_file: T::Boolean).void }
+        def add_proto_file(name, content, require_file: true)
+          add_content_file(tmp_path("proto/#{name}.proto"), content).tap do |proto_path|
```

Comment:
Ah, I just noticed something that might break using `add_content_file` like this. The `name` argument to `add_content_file` is used to build a tmp path under the `lib` folder, so when we supply a full temp path argument to the method now, it will add it under `lib` somewhere.

So, I think we might have either:
1. extract a `add_tmp_file` method that accepts a full path (not just a `name`) and use it here, or
2. check if the passed argument is already a temp path inside `add_content_file` and change behaviour accordingly.

I think 1 should be cleaner.

Reply:
Oh yeah it was weird. I ended up removing the `tmp_path` argument to `add_content_file` since it's already creating a tmp directory for us and `tmp_path` method returns the first created tmp directory. So now we have this starting with a temporary path.

```
λ tree
.
└── lib
    ├── content_pb.rb
    └── proto
        └── content.proto

3 directories, 2 files
```

Do you prefer `proto` being outside of `lib`?

Reply:
No, actually that looks good to me, thank you!